### ISSUE TITLE
well. it's a good code. but, it doesn't work on android x... :(

### ISSUE THREAD

**Issue by hub-DC** [KevinCS17/cobacoba#1]
Hi! few days age, we found your android project on github.
we were making tensorflow lite foreground service.
when we found your code, we were very happy!
Unfortunately, we didnt make it on android X.
samsung galaxy s7, we can build your code, and it successfully worked.
but, galaxy a21s( android x), foreground service doesnt work well.
if you know the answer, would you tell me how to work well?

Our team is suffering becuz of this problem... x(

Please help us... 
Our email is moneyrope2@gmail.com

**Comment by KevinCS17** [comment]
well sorry for late replying, I dont touch this program anymore since my undergraduate thesis done :(
something wrong with my program is, the apps only work for Camera API 2 for the Foreground Service

this is my story
firstly, i try this program on 2 phone Samsung Galaxy S10 (2019) and Samsung Galaxy S6 (2015)
I also tried in Samsung Galaxy A9 My brother phone..
and the foreground cant work..

why?
because Samsung Galaxy S6 and S10 use Camera API2
and A9 is Camera API 1 (LEGACY) or 1.5

my foreground code only for CAMERA API2
thats why it dont works..

your S7 iam pretty sure that CAMERA API2
thats why it can work

sorry for my bad english
i dont touch this code again, this code is public for my lecture request :))

it feels free if you want to used it
you can upgrade it to CAMERA API 1 and ANDROID JETPACK :)
my confusion is

why A9 that is newer phone than S6, has the lower API?

sorry for very late reply
thanks for the issue, and enjoy the progress
@hub-DC

**Comment by KevinCS17** [comment]
You can find my journal here, but its in Indonesian Language..

[Voice Alert as A Visual Assitant ](http://publication.petra.ac.id/index.php/teknik-informatika/article/view/10938)
@hub-DC 
thanks..

**Comment by hub-DC** [comment]
thank you!
some days later, i solved that problem.
i dont memorize what it is.
i edit "if" code in camera activity.
we are surprising about your code in now days.
thank you for your replying!
i tested this code in galaxy s7 so many times.
but, it used camera1 api not camera2 api.
my team was surprised too much...
so, we decided not to use camera1 api.
anyway, thank you for replying


2021년 6월 11일 (금) 오후 7:32, KevinCS17 ***@***.***>님이 작성:

**Comment by hub-DC** [comment]
and would you tell me how to train tensorflow?
im using roboflow but, i can train only 1000 images for free.
your tflite model is good... but, we want to retrain the model.
im using ssd_mobilenet_v2!

thank you!

2021년 6월 11일 (금) 오후 7:32, KevinCS17 ***@***.***>님이 작성:

**Comment by KevinCS17** [comment]
well i think many tutorial at youtube that help me to train that @hub-DC 
i ll search it later, and i ll mention you later.. @hub-DC

**Comment by KevinCS17** [comment]
https://www.youtube.com/watch?v=IOI0o3Cxv9Q&ab_channel=NicholasRenotte
this tutorial is for train in tensorflow2

there are 2 option in tensorflow 2
thats is create your own model or use pretrained model
if you like to create your own model, you can use tensroflow keras ( but i never try it)
if your like to retrain use pre trained model, the tutorial up there is the best i think

https://www.youtube.com/watch?v=oqd54apcgGE , this are good too..
https://www.youtube.com/watch?v=cvyDYdI2nEI , this is where i find google colab template

@hub-DC

**Comment by KevinCS17** [comment]
i just remembered that there already some application like this

[SPOT AND TELL CAM](https://play.google.com/store/apps/details?id=eu.sisik.camtospeech&hl=in&gl=US)
[FOREGROUND SERVICE GITHUB](https://github.com/sixo/background-cam)

@hub-DC

**Comment by hub-DC** [comment]
Thanks for the tip!
thank you for helping me,,,

i trained some cars, person and so on. but, it doesnt work well...
when i used for person, it told me motorcycle :(
i want to train well like you...
thanks a lot!!!!!!

2021년 6월 25일 (금) 오후 9:47, KevinCS17 ***@***.***>님이 작성:

**Comment by KevinCS17** [comment]
ahh I see

the database will effect the model knowledge, and..
you need to see how much learning rate and step you used to train the model

you check my paper in indonesia language, in chapter V for in table 5.7 page 130 [Voice ALert ](https://dewey.petra.ac.id/catalog/digital/preview?id=2199045)
the parameter I used are steps, and learning rate

You can see more detail on Table 5.3, 5.4, 5.5, 5.6 there on some page before

@hub-DC happy coding..

**Comment by hub-DC** [comment]
Thanks for the tip! when you come south korea, please visit us! we hope you
happy!!

2021년 6월 26일 (토) 오후 7:10, KevinCS17 ***@***.***>님이 작성:

**Comment by KevinCS17** [comment]
Youre wellcome, I really want to go there :(, I am South Korea fans : (( Running man, Twice, Tzuyu <3. Hope we can meet soon : ))

Best Of Luck and Stay Safe @hub-DC  :)

**Comment by hub-DC** [comment]
Hi! Long time no see! I ran deep learning training with 45,000 pictures of
cars, people, motorcycles, bicycles, and kickboards, but the performance is
not as good as the model you made. Can you give me your data set if you
need me? I'm begging you, please. When you come to Korea, I will treat you
with great hospitality.😉😉😉

2021년 7월 26일 (월) 오후 9:52, KevinCS17 ***@***.***>님이 작성:

**Comment by KevinCS17** [comment]
Well.. I remembered my friend and teacher said that too many image will make my model more dumb or stupid and that time i uses 1000 each class so total is 5000..

So what I did is.. I correcting my dataset choose which picture is good and not.. until it left 100 each class..

My Dataset are Google Open Image Dataset V6 that have all what I need except Stop Sign.

What I can suggest are
- Selecting picture which are good and delete which is bad..
-  Decrease the Image.. try 1000 500 250 or 100.. from my experience.. the 1000 give with only 5 class is too much.. and my genius friend told me so and to decrease it until 100 for training 25 image for testing (in train).
-  do not too much train the model.. how much step did you use? My research did 3 diffrent steo in 10000 12000 and 14000, the best model is in 10000.. my teacher said that 10000 is still overfitting but its okay.. you need to do more experiment in this section..
- try to reduce the learning rate to 0.01 0.1 0.001.. from my experience , it will effect the model too...


The last.. is some quotes from my teacher

"Machine learning / deep learning are 75% lucky  and 25% skill.."

So you need to do many expereince in tuning it


Goodluck

**Comment by KevinCS17** [comment]
> ahh I see
> 
> the database will effect the model knowledge, and..
> you need to see how much learning rate and step you used to train the model
> 
> you check my paper in indonesia language, in chapter V for in table 5.7 page 130 [Voice ALert ](https://dewey.petra.ac.id/catalog/digital/preview?id=2199045)
> the parameter I used are steps, and learning rate
> 
> You can see more detail on Table 5.3, 5.4, 5.5, 5.6 there on some page before
> 
> @hub-DC happy coding..

You can find my research in here.. @hub-DC 
Best of luck and stay safe :)

**Comment by hub-DC** [comment]
thank you so much for your help!!!!!!!😍😍😍😍😍😍
watch out covid-19!!!
good luck and happy coding~~!😚😚😚

2021년 8월 5일 (목) 오후 6:52, KevinCS17 ***@***.***>님이 작성:

**Comment by KevinCS17** [comment]
Youre Wellcome
Goodluck : ))

you can follow me on instagram @kevinchristiansalim (its okay if not) : ))
Dm Me for Followback :))

@hub-DC Goodluck..
